### ISSUE TITLE
IPv6 host: OpenError: The operation failed for reasons unrelated to the database itself and not covered by any other error code.

### ISSUE THREAD

**Issue by pruflyos** [ipfs/js-ipfs#2017]
- **Version**: 0.35.0
- **Platform**: Firefox 66.0.3 (64-bit)
- **Subsystem**:

#### Description:
I have a OpenWrt uHTTPd server which is reachable on both IPv4 and IPv6 address. If I put the example file from https://github.com/ipfs/js-ipfs/tree/master/examples/browser-script-tag on that server and access it via IPv4 (http://192.168.1.1/jsipfstest.html) everything is fine. But if I access the same file via IPv6 (http://[6001:3ff0:6dae:0:4819:b1bb:c001:22ac]/jsipfstest.html) the node is not started and the following error is shown in the browser console:
```
OpenError: The operation failed for reasons unrelated to the database itself and not covered by any other error code.
```
#### Steps to reproduce the error:
1. Have a webserver which is reachable on both - IPv4 and IPv6 address.
2. Put the example from https://github.com/ipfs/js-ipfs/tree/master/examples/browser-script-tag on that webserver.
3. Access the example via IPv4. It works!
4. Access the example via IPv6. It doesn't work!

**Comment by alanshaw** [comment]
Was there a stack trace with that error?

**Comment by alanshaw** [comment]
I do not get that error but this works for me in Firefox:

<img width="1072" alt="Screenshot 2019-05-14 at 09 58 14" src="https://user-images.githubusercontent.com/152863/57685062-f4413c80-762e-11e9-855a-be553b40d7c8.png">

...but is not considered secure in Chrome and I think we're not properly detecting secure context so webcrypto is failing:

<img width="1072" alt="Screenshot 2019-05-14 at 09 58 11" src="https://user-images.githubusercontent.com/152863/57685591-2010f200-7630-11e9-82f9-ddb44f0564bb.png">

@lidel have you seen this before? Shall we open an issue on `libp2p-crypto`?

**Comment by lidel** [comment]
@alanshaw it is a known issue with WebCrypto: it requires [Secure Context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (deploy over HTTPS or localhost IP). Firefox does not implement secure context requirement for it yet, but there is an [open issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1333140) to do it. 

**tl;dr** to test this without triggering secure context check load page from localhost IP:
```
http://[::1]:8000/ipfs.html
```

Upstream issues:
- https://github.com/libp2p/js-libp2p-crypto/issues/105
- https://bugzilla.mozilla.org/show_bug.cgi?id=1333140

**Comment by alanshaw** [comment]
A better error message will be displayed in future versions of `libp2p-crypto` https://github.com/libp2p/js-libp2p-crypto/pull/157

At some point we'll allow passing your own crypto implementation when in insecure contexts (for when you really really still need to do crypto), but that requires a big refactor of the module and of many modules that use it.

Considering this particular problem solved for now!
